### ISSUE TITLE
Toolbar Options not visible when Window size is small.

### ISSUE THREAD

**Issue by DarkPrince304** [zaproxy/zaproxy#2327]
Hello,

![screenshot from 2016-03-15 17 24 11](http://i.imgur.com/PF0XHSt.png)

As is visible from the above screenshot, many options in the Main Menu Toolbar and in the Active Scan Toolbar, etc are not visible. I would like to confirm this from someone. And if confirmed I would like to look in this :)

Thanks!

**Comment by psiinon** [comment]
I'm sure that will be the case.
How do you think it should be fixed? By having 2 lines for the toolbar in this case?
Its tricky because theres a lot of info to show, so handling very small screens is tricky.
And note that you should be able to run all of the tool bar options via the menus too, so it shouldnt stop you from being able to do anything (hopefully;)

**Comment by DarkPrince304** [comment]
Well I think it should be like how Mozilla Firefox browser works when there are too many tabs open :)

There's an arrow on both ends of the Tabs bar. Have a look here:
![screenshot from 2016-03-15 17 36 55](https://cloud.githubusercontent.com/assets/9005407/13777203/acd6adc8-ead4-11e5-9252-775127a32985.png)

**Comment by psiinon** [comment]
Thats tabs rather than a toolbar, so subtly different ;)
I'm not aware of any Swing controls which support these sort of arrows on toolbars, so you might have to create your own Swing component.

**Comment by DarkPrince304** [comment]
Okay, We can also have a drop down from a button at the right most corner like in Sublime Text?

**Comment by psiinon** [comment]
That would work, but not really sure how you'd do that either :P

**Comment by DarkPrince304** [comment]
Is there a super class for all the toolbars which all the types of toolbars extend?

**Comment by thc202** [comment]
Yes, but it's the Swing class, JToolBar. You would have to change all the codebase to use the new class (ZapToolBar?).
